lib/index.js

```diff
@@ -327,47 +374,56 @@ module.exports = class Breakpoints {
    * Convert a given string to a PascalCaseish format (without using a
    * dictionary is basically impossible)
    *
-   * @param {string} string -
-   * @returns {string}
+   * @static
+   * @param {string} string - String to convert to pascalish
+   * @returns {string} - Pascalish string
    */
-  static convertToPascalCaseish(string) {
+  static normalizeStrings(string) {
     return string
       .replace(/\s+/g, "")
       .split("_")
-      .map((subString) =>
-        subString
-          .replace(
-            /\w+/g,
-            (word) => word[0].toUpperCase() + word.slice(1).toLowerCase()
-          )
-          .replace(".", "-")
-      )
+      .map((subString) => convertToPascalCase(subString))
       .join("");
+
+    /**
+     * Converts to PascalCase
+     *
+     * @param {string} subString
+     * @returns {string}
+     */
+    function convertToPascalCase(subString) {
+      return subString
+        .replace(
+          /\w+/g,
+          (word) => word[0].toUpperCase() + word.slice(1).toLowerCase()
+        )
+        .replace(".", "-");
+    }
   }
 
   /**
    * Writes the css/js file
    *
-   * @param {string} filePath -
-   * @param {string} content -
-   * @param {Function} handleError -
+   * @static
+   * @async
+   * @param {string} filePath - FilePath
+   * @param {string} content - Content to write to file
+   * @returns {Promise} - Promise
    */
-  static writeFile(filePath, content, handleError) {
-    fs.writeFile(filePath, content, (error) => {
-      if (error !== null) {
-        handleError();
-      }
-    });
+  static async writeFile(filePath, content) {
+    await fs.promises.mkdir(path.dirname(filePath), { recursive: true });
+    return fs.promises.writeFile(filePath, content);
```

Comment:
Instead of using `fs.promises.…` you could also `const fs = require('node:fs/promises')` to get the promisified fs module directly.

(Then you would also need to move `readAndParseYamlConfiguration` to the async fs API. Which may not be that bad of an idea anyway.)